src/mqt/qecc/codes/css_code.py

```diff
@@ -25,79 +26,59 @@ def __init__(
-        return hash(x_hash ^ z_hash)
-
-    def __eq__(self, other: object) -> bool:
-        """Check if two CSS codes are equal."""
-        if not isinstance(other, CSSCode):
-            return NotImplemented
-        if self.Hx is None and other.Hx is None:
-            assert self.Hz is not None
-            assert other.Hz is not None
-            return np.array_equal(self.Hz, other.Hz)
-        if self.Hz is None and other.Hz is None:
-            assert self.Hx is not None
-            assert other.Hx is not None
-            return np.array_equal(self.Hx, other.Hx)
-        if (self.Hx is None and other.Hx is not None) or (self.Hx is not None and other.Hx is None):
-            return False
-        if (self.Hz is None and other.Hz is not None) or (self.Hz is not None and other.Hz is None):
-            return False
-        assert self.Hx is not None
-        assert other.Hx is not None
-        assert self.Hz is not None
-        assert other.Hz is not None
-        return bool(
-            mod2.rank(self.Hx) == mod2.rank(np.vstack([self.Hx, other.Hx]))
-            and mod2.rank(self.Hz) == mod2.rank(np.vstack([self.Hz, other.Hz]))
-        )
+    def x_checks_as_pauli_strings(self) -> list[str]:
+        """Return the x checks as Pauli strings."""
+        return ["".join("X" if bit == 1 else "I" for bit in row) for row in self.Hx]
 
-    @staticmethod
-    def _compute_logical(m1: npt.NDArray[np.int8] | None, m2: npt.NDArray[np.int8] | None) -> npt.NDArray[np.int8]:
-        """Compute the logical matrix L."""
-        if m1 is None:
-            ker_m2 = mod2.nullspace(m2)  # compute the kernel basis of m2
-            pivots = mod2.row_echelon(ker_m2)[-1]
-            logs = np.zeros_like(ker_m2, dtype=np.int8)  # type: npt.NDArray[np.int8]
-            for i, pivot in enumerate(pivots):
-                logs[i, pivot] = 1
-            return logs
+    def z_checks_as_pauli_strings(self) -> list[str]:
+        """Return the z checks as Pauli strings."""
+        return ["".join("Z" if bit == 1 else "I" for bit in row) for row in self.Hz]
 
-        if m2 is None:
-            return mod2.nullspace(m1).astype(np.int8)
+    def x_logicals_as_pauli_strings(self) -> list[str]:
+        """Return the x logicals as a Pauli strings."""
+        return ["".join("X" if bit == 1 else "I" for bit in row) for row in self.Lx]
 
+    def z_logicals_as_pauli_strings(self) -> list[str]:
+        """Return the z logicals as Pauli strings."""
+        return ["".join("Z" if bit == 1 else "I" for bit in row) for row in self.Lz]
+
+    @staticmethod
+    def _compute_logical(m1: npt.NDArray[np.int8], m2: npt.NDArray[np.int8]) -> npt.NDArray[np.int8]:
+        """Compute the logical matrix L."""
         ker_m1 = mod2.nullspace(m1)  # compute the kernel basis of m1
         im_m2_transp = mod2.row_basis(m2)  # compute the image basis of m2
         log_stack = np.vstack([im_m2_transp, ker_m1])
```

Comment:
looking at `get_x_syndrome` etc. I think it would  be good to check naming conventions. usually I like to call the syndrome the same as the error so the x syndrome comes from an x error but is computed with Hz etc. does that make sense? anyways being consistent e.g., with the logicals would be good

Reply:
Oh yeah, I didn't even think of that... I will change that.